src/test/java/org/opensearch/knn/index/FaissIT.java

```diff
@@ -151,6 +151,181 @@ public void testEndToEnd_fromMethod() throws Exception {
         fail("Graphs are not getting evicted");
     }
 
+    public void testEndToEnd_fromNSGMethod() throws Exception {
+        String indexName = "test-index-1";
+        String fieldName = "test-field-1";
+
+        KNNMethod nsgMethod = KNNEngine.FAISS.getMethod(KNNConstants.METHOD_NSG);
+        SpaceType spaceType = SpaceType.L2;
+
+        List<Integer> mValues = ImmutableList.of(16, 32, 64, 128);
+        List<Integer> efConstructionValues = ImmutableList.of(16, 32, 64, 128);
+        List<Integer> efSearchValues = ImmutableList.of(16, 32, 64, 128);
+
+        Integer dimension = testData.indexData.vectors[0].length;
+
+        // Create an index
+        XContentBuilder builder = XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject("properties")
+            .startObject(fieldName)
+            .field("type", "knn_vector")
+            .field("dimension", dimension)
+            .startObject(KNNConstants.KNN_METHOD)
+            .field(KNNConstants.NAME, nsgMethod.getMethodComponent().getName())
+            .field(KNNConstants.METHOD_PARAMETER_SPACE_TYPE, spaceType.getValue())
+            .field(KNNConstants.KNN_ENGINE, KNNEngine.FAISS.getName())
+                .startObject(KNNConstants.PARAMETERS)
+                .field(KNNConstants.METHOD_PARAMETER_M, 64)
+                .endObject()
+            .endObject()
+            .endObject()
+            .endObject()
+            .endObject();
+
+        Map<String, Object> mappingMap = xContentBuilderToMap(builder);
+        String mapping = Strings.toString(builder);
+
+        createKnnIndex(indexName, mapping);
+        Map<String, Object>  maping = getIndexMappingAsMap(indexName);
+//        assertEquals(new TreeMap<>(mappingMap), new TreeMap<>(maping));
+
+        // Index the test data
+        bulkAddKnnDocs(indexName, fieldName, testData.indexData.vectors, testData.indexData.docs.length);
+//        for (int i = 0; i < testData.indexData.docs.length; i++) {
+//            addKnnDoc(
+//                indexName,
+//                Integer.toString(testData.indexData.docs[i]),
+//                fieldName,
+//                Floats.asList(testData.indexData.vectors[i]).toArray()
+//            );
+//        }
```

Comment:
Please remove all the commented code from the PR.

Reply:
i try to explain the scenario. i'll remove it
>            /** TODO:
>            * when numIds is too small, NSG graph would core/throw exception
>             * because There are too much invalid entries in the knn graph.
 >            */

Reply:
https://github.com/opensearch-project/k-NN/pull/945/commits/819fcedce70fc452fb55152ab873a0e5dea53923 removed all commented code